apps/dashboard/src/components/workflow-editor/steps/configure-inline/configure-step-inline-form.tsx

```diff
@@ -0,0 +1,94 @@
+import { useCallback, useEffect, useMemo } from 'react';
+import merge from 'lodash.merge';
+import { useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import type { StepDataDto, StepIssuesDto, UpdateWorkflowDto, WorkflowResponseDto } from '@novu/shared';
+import { StepTypeEnum } from '@novu/shared';
+
+import { flattenIssues, updateStepControlValuesInWorkflow } from '@/components/workflow-editor/step-utils';
+import { buildDefaultValues, buildDefaultValuesOfDataSchema, buildDynamicZodSchema } from '@/utils/schema';
+import { Form } from '@/components/primitives/form/form';
+import { useFormAutosave } from '@/hooks/use-form-autosave';
+import { SaveFormContext } from '@/components/workflow-editor/steps/save-form-context';
+import { DelayForm } from '@/components/workflow-editor/steps/delay/delay-form';
+
+const STEP_TYPE_TO_INLINE_FORM: Record<StepTypeEnum, (args: StepInlineFormProps) => React.JSX.Element | null> = {
+  [StepTypeEnum.DELAY]: DelayForm,
+  [StepTypeEnum.IN_APP]: () => null,
+  [StepTypeEnum.EMAIL]: () => null,
+  [StepTypeEnum.SMS]: () => null,
+  [StepTypeEnum.CHAT]: () => null,
+  [StepTypeEnum.PUSH]: () => null,
+  [StepTypeEnum.CUSTOM]: () => null,
+  [StepTypeEnum.TRIGGER]: () => null,
+  [StepTypeEnum.DIGEST]: () => null,
+};
+
+// Use the UI Schema to build the default values if it exists else use the data schema (code-first approach) values
+const calculateDefaultValues = (step: StepDataDto) => {
+  if (Object.keys(step.controls.uiSchema ?? {}).length !== 0) {
+    return merge(buildDefaultValues(step.controls.uiSchema ?? {}), step.controls.values);
+  }
+
+  return merge(buildDefaultValuesOfDataSchema(step.controls.dataSchema ?? {}), step.controls.values);
+};
+
+export type StepInlineFormProps = {
+  workflow: WorkflowResponseDto;
+  step: StepDataDto;
+};
+
+type ConfigureStepInlineFormProps = StepInlineFormProps & {
+  issues?: StepIssuesDto;
+  update: (data: UpdateWorkflowDto) => void;
+  updateStepCache: (step: Partial<StepDataDto>) => void;
+};
+
+export const ConfigureStepInlineForm = (props: ConfigureStepInlineFormProps) => {
+  const { workflow, step, issues, update, updateStepCache } = props;
```

Comment:
we are drilling the props quite deep, we should think about using the hooks here, but I left it like this for consistency.